### PR TITLE
GA the data stream lifecycle

### PR DESCRIPTION
**Pull request by andreidan**
This makes the data stream lifecycle generaly available. This will allow data streams to take advantage of a native simplified and resilient lifecycle implementation.

This will be followed-up with PRs on enhancing documentation on:
- automatic force merge
- default managed data stream 
- mixing ILM and data stream lifecycle together with how to migrate between to two systems

### CODE REVIEW

**Comment by andreidan** [comment]
Failure is https://github.com/elastic/elasticsearch/issues/98539

**Comment by andreidan** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.get_data_lifecycle.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
       "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/data-streams-get-lifecycle.html",
       "description":"Returns the data stream lifecycle of the selected data streams."
     },
-    "stability":"experimental",
+    "stability":"stable",
```
Comment (gmarouli) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.get_data_lifecycle.json:7]: Good catch, after this is merged we need to also update the rest-spec.

**Review by gmarouli — APPROVED** [review]
Thank you for this @andreidan for this! I am so happy to see it GA 🚀

File: docs/changelog/98644.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  body: |-
+    This makes the data stream lifecycle generaly available. Data streams will 
+    be able to take advantage of a built-in simplified and resilient
+    lifecycle implementation. Data streams with a configured lifecycle will
+    be automatically rolled over and tail merged (a forcemerge implementation 
+    that's lightweight and only merges the long tail of small segments instead 
+    of the whole shard). With the shard and index maintenance tasks being 
+    handled automatically to ensure optimum performance, and trade-off between
+    indexing and searching, you'll be able to focus on the business related 
+    lifecycle aspects like data retention.
```
[the remark this answers is not in the document]
Comment (andreidan) [reply docs/changelog/98644.yaml]: Sure, done

**Comment by andreidan** [comment]
The work here is superseded by https://github.com/elastic/elasticsearch/pull/100187

**Pull request merged**